sessions/test_js_challenge.py

```diff
@@ -325,3 +338,193 @@ def test_fail_challenge_too_late(self):
                                   delay_min=1000, delay_range=1000,
                                   status_code=503, expect_pass=False,
                                   req_delay=3, restart_on_fail=True)
+
+    def client_send_custom_req(self, client, uri, cookie):
+        req = ("GET %s HTTP/1.1\r\n"
+               "Host: localhost\r\n"
+               "%s"
+               "\r\n" % (uri, ("Cookie: %s=%s\r\n" % cookie) if cookie else ""))
+        response = self.client_send_req(client, req)
+
+        self.assertEqual(response.status,'302',
+                         "unexpected response status code")
+        c_header = response.headers.get('Set-Cookie', None)
+        self.assertIsNotNone(c_header,
+                             "Set-Cookie header is missing in the response")
+        match = re.search(r'([^;\s]+)=([^;\s]+)', c_header)
+        self.assertIsNotNone(match,
+                             "Cant extract value from Set-Cookie header")
+        cookie = (match.group(1), match.group(2))
+
+        uri = response.headers.get('Location', None)
+        self.assertIsNotNone(uri,
+                             "Location header is missing in the response")
+        return (uri, cookie)
+
+    def client_send_first_req(self, client, uri):
+        return self.client_send_custom_req(client, uri, None)
+
+    def test_disable(self):
```

Comment:
`test_disable` -> `test_disable_challenge_on_reload`? Not trying to be nitpick, but usually good test names much more useful  - it's easy to understand, which test failed without looking into the sources.

Reply:
BTW, the test is close to `JSChallengeAfterReload`. It worth to group tests into classes by feature-under-test, i.e. put all reload-related into the same class. In most cases it's possible. It gives you an opportunity to run all tests related to a single feature with a single argument. I always use such opportunities during development.